src/mmds/src/lib.rs

```diff
@@ -65,6 +65,20 @@ fn build_response(http_version: Version, status_code: StatusCode, body: Body) ->
     response
 }
 
+// Make the URI a correct JSON pointer value.
+fn sanitize_uri(uri: &str) -> String {
+    let mut sanitized_uri = uri.to_owned();
+    let mut len = u32::MAX as usize;
+    // Loop while the deduping decreases the sanitized len.
+    // Each iteration will attempt to dedup "//".
+    while sanitized_uri.len() < len {
+        len = sanitized_uri.len();
+        sanitized_uri = sanitized_uri.replace("//", "/");
```

Comment:
Does it make sense to use a pattern and replace all the occurrences of "//+" with "/"? This wouldn't involve a while loop to reduce the "//" resulted from previous reduction in steps.

Reply:
I was searching that too but it seems rust doesn't offer support for regex in the std library.

Reply:
That sounds like we need to pull in some extra dependency to use regular expressions. This code is not on any hotpath so I do not want to make such trade-offs,

Reply:
In this case, I am OK with the while loop.